File: OptiProFrontend/src/app/models/message-block.ts
/**
 * @author Michael Banck michael.banck@uni-wuerzburg.de ©
 */
export class MessageBlock {
  constructor(public systemInfoS: string, public messages: ChatMessage[]) {
  }

}

export class ChatMessage {
  constructor(public text: string, public isBot: boolean = false, public link: string = "") {
  }
}

File: ChatGptBackend/MessageBlock.py
from pydantic import BaseModel
from typing import List


class ChatMessage(BaseModel):
    """
    @author Michael Banck michael.banck@uni-wuerzburg.de ©
    """
    text: str
    isBot: bool


class MessageBlock(BaseModel):
    """
    @author Michael Banck michael.banck@uni-wuerzburg.de ©
    """
    systemInfo: str
    messages: List[ChatMessage]
